python/lsst/ts/scheduler/scheduler_csc.py

```diff
@@ -1579,3 +1676,277 @@ def callback_script_info(self, data):
+
+        driver_type = None
+        for member in members_of_driver_lib:
+            try:
+                if issubclass(member[1], Driver):
+                    self.log.debug("Found driver %s%s", member[0], member[1])
+                    driver_type = member[1]
+            except TypeError:
+                pass
+
+        if driver_type is None:
+            raise RuntimeError("Could not find Driver on module %s" % driver_type)
+
+        self.driver = driver_type(
+            models=self.models, raw_telemetry=self.raw_telemetry, log=self.log
+        )
+
+    async def _handle_startup_database_snapshot(self, startup_database: str) -> None:
+        """Handle startup database snapshot.
+
+        Parameters
+        ----------
+        startup_database : str
+            Uri of the startup database.
+
+        Raises
+        ------
+        RuntimeError
+            If startup_database is invalid.
+        """
+
+        if is_uri(startup_database):
+            self.log.info(f"Loading scheduler snapshot from {startup_database}.")
+            await self._handle_load_snapshot(startup_database)
+        elif startup_database.strip():
+            raise RuntimeError(
+                f"Invalid startup_database: {startup_database.strip()}. "
+                "Make sure it is a valid and accessible URI."
+            )
+        else:
+            self.log.debug("No scheduler snapshot provided.")
+
+    async def _handle_startup_database_observation_db(
+        self, startup_database: str
+    ) -> None:
+        """Handle startup database for observation database.
+
+        Parameters
+        ----------
+        startup_database : str
+            Path to a local database or EFD query.
+
+        Raises
+        ------
+        RuntimeError
+            If startup_database is invalid.
+        """
+
+        if not startup_database.strip():
+            self.log.info("No observation history information provided.")
+        elif pathlib.Path(startup_database).exists():
+            self.log.info(
+                f"Loading observation history from database: {startup_database}."
+            )
+            await self._handle_load_observations_from_db(database_path=startup_database)
+        elif is_valid_efd_query(startup_database):
+            await self._handle_load_observations_from_efd(efd_query=startup_database)
+        else:
+            error_msg = (
+                "Specified startup database does not exists and does not classify as an EFD query. "
+                f"Received: {startup_database}. "
+                "If this was supposed to be a path, it must be local to the CSC environment. "
+                "If this was supposed to be an EFD query, it should have the format: "
+                'SELECT * FROM "efd"."autogen"."lsst.sal.Scheduler.logevent_observation" WHERE '
+                "time >= '2021-06-09T00:00:00.000+00:00' AND time <= '2021-06-12T00:00:00.000+00:00'"
```

Comment:
Strictly speaking an EFD query can also use times in the format '2021-06-09T00:00:00Z' and possibly other formats as well so this message is perfectly OK as it is.